ksql-engine/src/main/java/io/confluent/ksql/analyzer/AggregateAnalyzer.java

```diff
@@ -36,21 +37,26 @@
   private final MutableAggregateAnalysis aggregateAnalysis;
   private final QualifiedColumnReferenceExp defaultArgument;
   private final FunctionRegistry functionRegistry;
+  private final boolean hasWindowExpression;
 
   AggregateAnalyzer(
       final MutableAggregateAnalysis aggregateAnalysis,
       final QualifiedColumnReferenceExp defaultArgument,
+      final boolean hasWindowExpression,
       final FunctionRegistry functionRegistry
   ) {
     this.aggregateAnalysis = Objects.requireNonNull(aggregateAnalysis, "aggregateAnalysis");
     this.defaultArgument = Objects.requireNonNull(defaultArgument, "defaultArgument");
     this.functionRegistry = Objects.requireNonNull(functionRegistry, "functionRegistry");
+    this.hasWindowExpression = hasWindowExpression;
   }
 
   void processSelect(final Expression expression) {
     final Set<ColumnReferenceExp> nonAggParams = new HashSet<>();
     final AggregateVisitor visitor = new AggregateVisitor((aggFuncName, node) -> {
-      if (!aggFuncName.isPresent()) {
+      if (aggFuncName.isPresent()) {
+        throwOnWindowBoundColumnIfWindowedAggregate(node);
```

Comment:
this is not really the right error message right? in this case the column is used in the select, but it's not supported because it's a parameter to an aggregate function.

Reply:
Error message submitted in next PR will be explicit about not being able to use window bounds as parameters to UDAFs.

Reply:
Fix in https://github.com/confluentinc/ksql/pull/4459